webapp/src/ts/modals/edit-user/update-password.component.ts

```diff
@@ -0,0 +1,123 @@
+import {EditUserAbstract} from "@mm-modals/edit-user/edit-user.component";
+
+const passwordTester = require('simple-password-tester');
+import {Component} from "@angular/core";
+import {UserSettingsService} from "@mm-services/user-settings.service";
+import {TranslateService} from "@ngx-translate/core";
+import {UpdateUserService} from "@mm-services/update-user.service";
+import {BsModalRef} from "ngx-bootstrap/modal";
+
+const PASSWORD_MINIMUM_LENGTH = 8;
+const PASSWORD_MINIMUM_SCORE = 50;
+
+
+@Component({
+  selector: 'update-password',
+  templateUrl: './update-password.component.html'
+})
+export class UpdatePasswordComponent extends EditUserAbstract {
+
+  editUserModel: {
+    username?,
+    currentPassword?,
+    password?,
+    passwordConfirm?
+  } = {};
+
+  errors: {
+    currentPassword?,
+    password?
+  } = {};
+
+  constructor(
+    bsModalRef: BsModalRef,
+    userSettingsService: UserSettingsService,
+    private translateService: TranslateService,
+    private updateUserService: UpdateUserService,
+  ) {
+    super(bsModalRef, userSettingsService);
+  }
+
+  updatePassword() {
+    this.setProcessing();
```

Comment:
It looks like you're missing the error reset part. This might mean that, once an error is shown, it remains shown for ever, even if the field values are changed.

Reply:
Yes you are right, I did test that after errors are displayed and fixed the form is submitted, but if I fix an error and perform a new one, the new one is displayed but the previous error is not cleared. I'll fix it.

Reply:
Fixed.